lib/Config.js

```diff
@@ -280,6 +280,46 @@ function requestsConfigAssert(requestsConfig) {
     }
 }
 
+function bucketNotifAssert(bucketNotifConfig) {
+    assert(Array.isArray(bucketNotifConfig),
+        'bad config: bucket notification configuration must be an array');
+    bucketNotifConfig.forEach(c => {
+        const { resource, type, host, port, auth } = c;
+        assert(typeof resource === 'string',
+            'bad config: bucket notification configuration resource must be a string');
+        assert(typeof type === 'string',
+            'bad config: bucket notification configuration type must be a string');
+        assert(typeof host === 'string' && host !== '',
+            'bad config: hostname must be a non-empty string');
+        assert(Number.isInteger(port) && port > 0,
+            'bad config: port must be a positive integer');
+        assert(typeof auth === 'object',
+            'bad config: bucket notification auth must be an object');
+        const { cert, user, password } = auth;
+        if ((cert && user && password) || (!cert && !user && !password)) {
+            throw new Error('bad config: bucket notification configuration ' +
+                'auth should contain either cert or user and password');
+        }
+        if (cert) {
+            const certpath = (cert[0] === '/') ? cert : `${this._basePath}/${cert}`;
+            assert.doesNotThrow(() =>
+                fs.accessSync(certpath, fs.F_OK | fs.R_OK),
+                `File not found or unreachable: ${certpath}`);
+            // eslint-disable-next-line no-param-reassign
+            c.auth.cert = certpath;
+        } else if (!(user && password)) {
```

Comment:
Not sure if this is any better looking or easier to read but you could also do:
```suggestion
        } else if (!user || !password) {
```